netbox_onboarding/onboard.py

```diff
@@ -12,551 +12,110 @@
-    """Used to maintain information about the network device during the onboarding process."""
-
-    def __init__(self, onboarding_task, username=None, password=None, secret=None):
-        """Initialize the network device keeper instance and ensure the required configuration parameters are provided.
-
-        Args:
-          onboarding_task (OnboardingTask): Task being processed
-          username (str): Device username (if unspecified, NAPALM_USERNAME settings variable will be used)
-          password (str): Device password (if unspecified, NAPALM_PASSWORD settings variable will be used)
-          secret (str): Device secret password (if unspecified, NAPALM_ARGS["secret"] settings variable will be used)
-
-        Raises:
-          OnboardException('fail-config'):
-            When any required config options are missing.
-        """
-        self.ot = onboarding_task
-
-        # Attributes that are set when reading info from device
-
-        self.hostname = None
-        self.vendor = None
-        self.model = None
-        self.serial_number = None
-        self.mgmt_ifname = None
-        self.mgmt_pflen = None
-        self.username = username or settings.NAPALM_USERNAME
-        self.password = password or settings.NAPALM_PASSWORD
-        self.secret = secret or settings.NAPALM_ARGS.get("secret", None)
-
-    def check_reachability(self):
-        """Ensure that the device at the mgmt-ipaddr provided is reachable.
-
-        We do this check before attempting other "show" commands so that we know we've got a
-        device that can be reached.
-
-        Raises:
-          OnboardException('fail-connect'):
-            When device unreachable
-        """
-        ip_addr = self.ot.ip_address
-        port = self.ot.port
-        timeout = self.ot.timeout
-
-        logging.info("CHECK: IP %s:%s", ip_addr, port)
-
-        try:
-            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            sock.settimeout(timeout)
-            sock.connect((ip_addr, port))
-
-        except (socket.error, socket.timeout, ConnectionError):
-            raise OnboardException(reason="fail-connect", message=f"ERROR device unreachable: {ip_addr}:{port}")
-
-    @staticmethod
-    def check_netmiko_conversion(guessed_device_type, platform_map=None):
-        """Method to convert Netmiko device type into the mapped type if defined in the settings file.
-
-        Args:
-            guessed_device_type (string): Netmiko device type guessed platform
-            test_platform_map (dict): Platform Map for use in testing
-
-        Returns:
-            string: Platform name
-        """
-        # If this is defined, process the mapping
-        if platform_map:
-            # Attempt to get a mapped slug. If there is no slug, return the guessed_device_type as the slug
-            return platform_map.get(guessed_device_type, guessed_device_type)
+class OnboardingTaskManager(object):
+    """Onboarding Task Manager."""
```

Comment:
Thinking about this after our discussion this morning -- does it actually make sense to have this class that essentially just offers a layer of abstraction atop `OnboardingTask`, or would it be better to just add the necessary functionality (at present just the `napalm_driver` property) directly to `OnboardingTask` and skip the definition of this class?

Reply:
Agreed it's a bit redundant now, although would keep it until implementations in the field, ok?